### DISCUSSION TITLE
Is using on-screen keyboard inside anvil possible?

### DISCUSSION THREAD

**Discussion by septatrix** [Smithay/smithay#902] [Q&A]
I am currently looking for a lightweight compositor for a kiosk application. [Cage](https://github.com/Hjdskes/cage) and weston's [`kiosk-shell.so`](https://wayland.pages.freedesktop.org/weston/toc/kiosk-shell.html) seemed great at first though the former does not yet support `wlr-layer-shell-v1` and I have not gotten the latter to work.

So my question is whether anvil already supports enabling an OSK, if it is planned for the future or if it is considered out of scope.

I feel like this would be an ideal use-case for anvil as kiosks do not need any fancy window management, transition etc. If it is not too hard I could also add this myself if someone guides me to where changes would be required.

**Comment by Drakulix** [comment]
We do support the virtual-keyboard protocol  (as can be see [here](https://github.com/Smithay/smithay/issues/781)), which gives you the option to use many third-party keyboards with smithay/anvil.

**Reply by septatrix** [reply]
Yes I can manually show the keyboard however it does not automatically pop up. `weston` with `weston-keyboard` configured automatically shows the keyboard when a text field is selected (at least in the demo `weston-editor`). GNOME's OSK is automatically shown for many modern applications (GNOME Terminal, Firefox, squeekboard-entry etc.).

Instead, if I start squeekboard inside anvil, it does not automatically pop up when selecting input fields. While I can manually bring it up by calling `busctl call --user sm.puri.OSK0 /sm/puri/OSK0 sm.puri.OSK0 SetVisible b true` this is not really useful as it would be infeasible to patch every application to call this (and also hide it, respect hints like numpad layout and similar stuff). I am not sure if there are some other things I would need to configure or if the compositor (anvil) has to detect whenever an input-field is selected and invoke an OSK somehow.
